master/internal/api_agents.go

```diff
@@ -4,18 +4,24 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/determined-ai/determined/master/internal/sproto"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
 	"github.com/determined-ai/determined/master/pkg/actor"
 	"github.com/determined-ai/determined/proto/pkg/apiv1"
 )
 
 func (a *apiServer) GetAgents(
-	_ context.Context, req *apiv1.GetAgentsRequest) (resp *apiv1.GetAgentsResponse, err error) {
-	response := a.m.system.AskAt(
-		actor.Addr("resourceManagers"), sproto.GetEndpointActorAddress{}).Get()
-	endpointActorName := response.(string)
-
-	err = a.actorRequest(endpointActorName, req, &resp)
+	_ context.Context, req *apiv1.GetAgentsRequest,
+) (resp *apiv1.GetAgentsResponse, err error) {
+	switch {
+	case a.m.system.Get(actor.Addr("agents")) != nil:
+		err = a.actorRequest("/agents", req, &resp)
+	case a.m.system.Get(actor.Addr("pods")) != nil:
```

Comment:
we leak implementation details here because the actor address we rely on is not consistent in different implementations (pods and agents). IMO if we have to be consistent, we should make the actor address of pods to be `agents`. Another approach is to define those APIs inside the `resourcemanagers` package rather than outside of it.